data/types/bolus/pen/pen_test.go

```diff
@@ -0,0 +1,159 @@
+package pen_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	dataNormalizer "github.com/tidepool-org/platform/data/normalizer"
+	"github.com/tidepool-org/platform/data/types/bolus"
+	"github.com/tidepool-org/platform/data/types/bolus/pen"
+	dataTypesBolusPenTest "github.com/tidepool-org/platform/data/types/bolus/pen/test"
+	"github.com/tidepool-org/platform/data/types/common"
+	dataTypesTest "github.com/tidepool-org/platform/data/types/test"
+	errorsTest "github.com/tidepool-org/platform/errors/test"
+	"github.com/tidepool-org/platform/pointer"
+	"github.com/tidepool-org/platform/structure"
+	structureValidator "github.com/tidepool-org/platform/structure/validator"
+)
+
+func NewMeta() interface{} {
+	return &bolus.Meta{
+		Type:    "bolus",
+		SubType: "pen",
+	}
+}
+
+var _ = Describe("Pen", func() {
+	It("SubType is expected", func() {
+		Expect(pen.SubType).To(Equal("pen"))
+	})
+
+	Context("New", func() {
+		It("returns the expected datum with all values initialized", func() {
+			datum := pen.New()
+			Expect(datum).ToNot(BeNil())
+			Expect(datum.Type).To(Equal("bolus"))
+			Expect(datum.SubType).To(Equal("pen"))
+			Expect(datum.Normal).To(BeNil())
+		})
+	})
+
+	Context("Pen", func() {
+		Context("Parse", func() {
+			// TODO
+		})
+
+		Context("Validate", func() {
+			DescribeTable("validates the datum",
+				func(mutator func(datum *pen.Pen), expectedErrors ...error) {
+					datum := dataTypesBolusPenTest.NewPen()
+					mutator(datum)
+					dataTypesTest.ValidateWithExpectedOrigins(datum, structure.Origins(), expectedErrors...)
+				},
+				Entry("succeeds",
+					func(datum *pen.Pen) {},
+				),
+				Entry("type missing",
```

Comment:
Not sure if it is necessary to test the presence of type/subType.
It is already done in the bolus_test & base_test.

Reply:
yep that makes the tests pretty verbose. I followed what was done in similar objects